sql/src/main/java/io/crate/execution/dsl/package-info.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+/**
+ * <p>
+ *  This package contains classes for the execution domain-specific language (DSL).
+ *  This DSL is used to describe how queries are executed.
+ * </p>
+ *
+ * <h2>Phases</h2>
+ *
+ * <p>
+ *  A ExecutionPhase is the "core" of the execution DSL. The interfaces are:
+ * </p>
+ * <ul>
+ *  <li>{@link io.crate.execution.dsl.phases.ExecutionPhase}</li>
+ *  <li>{@link io.crate.execution.dsl.phases.UpstreamPhase}</li>
+ * </ul>
+ * <p>
+ *  There are three categories of ExecutionPhases:
+ * </p>
+ *
+ * <ul>
+ *  <li>
+ *      Phases that describes *how* to produce *what* data<br />
+ *      ({@link io.crate.execution.dsl.phases.CollectPhase})
+ *  </li>
+ *  <li>
+ *      Phases that describes how to receive receive and process data
+ *      ({@link io.crate.execution.dsl.phases.MergePhase}, {@link io.crate.execution.dsl.phases.NestedLoopPhase})
+ *  </li>
+ *  <li>
+ *      Phases that instruct the executor to create some kind of "context" which may be used by other phases.
+ *      ({@link io.crate.execution.dsl.phases.FetchPhase})
+ *  </li>
+ * </ul>
+ *
+ * <p>
+ *  Data generally "flows through" phases. For example:
+ * </p>
+ * <pre>
+ *     select * from t1
+ *
+ *     Would be represented as
+ *
+ *     CollectPhase (Running on [node1, node2, node3])
+ *          |
+ *     MergePhase (Running on [node1])
+ *
+ *     The data flows from CollectPhase to MergePhase
+ * </pre>
+ * <p>
+ *     This data flow is implicit defined in a concrete execution plan due to the order how the phases appear in the plan.
+ *     {@link io.crate.execution.dsl.phases.NodeOperation} is more explicit in that regard and is used in the Executor.
```

Comment:
NodeOperation holds the ExecutionPhase during execution.

What is the Executor? You mean `io.crate.executor.*`?

Reply:
What do you mean with `NodeOperation holds the ExecutionPhase` during execution? That the NodeOperation contains a ExecutionPhase is only 1 aspect of it, but not everything. That's why I generalized to `is used`.  `Executor` refers generically to all components involved in the execution. There are a couple of those.

Maybe I could re-phrase it to `is more explicit in that regard and is used during query execution` ?

Reply:
It is hard to understand in what regard it is more explicit. 

How about: `The NodeOperation class is used to represent the operation and the data flow during runtime.`